dask_kubernetes/operator/daskcluster.py

```diff
@@ -82,14 +82,15 @@ def build_scheduler_service_spec(name):
     }
 
 
-def build_worker_pod_spec(name, namespace, image, n):
+def build_worker_pod_spec(name, namespace, image, n, scheduler_name):
     return {
         "apiVersion": "v1",
         "kind": "Pod",
         "metadata": {
             "name": f"{name}-worker-{n}",
```

Comment:
I think this name should also include the `scheduler_name` otherwise we will have collisions with multiple clusters.